### ISSUE TITLE
Question on the official dataset splits

### ISSUE THREAD

**Issue by JunMa11** [vanvalenlab/intro-to-deepcell#43]
Dear Vanvalenlab,

Thanks for sharing the great work.

I download the TissueNet data from https://datasets.deepcell.org/. The data splits are training (2601), validation (3140), testing (1249), which is different from the description in the paper:

>  the data were randomly split into training (80%), validation (10%) and testing (10%) fractions

Is there anything that I missed?

BTW, do the [panopticnet](https://github.com/vanvalenlab/deepcell-tf/blob/master/deepcell/model_zoo/panopticnet.py) mean `Mesmer` in the paper?
https://www.nature.com/articles/s41587-021-01094-0#Sec3

**Comment by willgraf** [comment]
I'll let @ngreenwald comment on the data splits.

Yes, `Mesmer` is an implementation of the `PanOpticNet` model using 2 channel inputs.

**Comment by JunMa11** [comment]
Hi @ngreenwald ,

Thanks for your reply very much.

1. During training, are the 256X256 patches generated by this function?
https://github.com/vanvalenlab/deepcell-tf/blob/master/deepcell/image_generators/cropping.py

2. In the Extended Data Fig. 2 c-d, are these scores based on the above testing patches (1249)?

![Extended Data Fig. 2 ](https://user-images.githubusercontent.com/19947331/149394717-0f89730d-032e-4e79-aa63-912628bba40c.png)

If the answer is yes, where can I find the corresponding segmentation results?

> I download a folder from https://storage.googleapis.com/publications-data/mesmer-paper/publication_data_folder.zip

how to understand the results in these folders? Each folder has multiple splits.

![1642101873(1)](https://user-images.githubusercontent.com/19947331/149395808-38791cd2-f863-4192-9d19-4f6755cef4ad.png)

3. Moreover, I also try to download this training data. The download process went well (56.2GB), but the unzip process failed. I have tried three times. Each time has the same error. What is the difference between this train data and the above training data (2601 512x512 patches)?
![1642102095(1)](https://user-images.githubusercontent.com/19947331/149396381-58c9919b-ca5c-46f1-9d10-29d706636bf5.png)

![1642102227(1)](https://user-images.githubusercontent.com/19947331/149396683-5f126cb5-4674-451d-97bb-d63044e5fe60.png)

Looking forward to your reply:)

Kindest regards,
Jun

**Comment by ngreenwald** [comment]
Hey Jun, see my answers below. Please let me know if any of that isn't clear

1. Yes, see here for an example of what that looks like to train a model https://github.com/vanvalenlab/publication-figures/blob/master/2021-Greenwald_Miller_et_al-Mesmer/Mesmer_training_notebook.ipynb
2. Yes, the folder you showed is the right area. If you notice from the barchart, there are three dots for each model. Those represent three replicates, where we performed a different random seed for creating the train/val/test splits. Hence, there are three 'splits' for each model, with the corresponding segmentation predictions for each. 
3. The training data in the large (56GB zip) contains the data specifically for training all of the different permutations of the model we show in the paper: the one that is only trained on breast cancer data, only trained on CODEX data, only trained with 1/1000th the number of cells, etc. If you're interested in training the model with the full dataset, then you don't need this extra data. I'll investigate to see what the issue is with the zip file

**Comment by JunMa11** [comment]
Hi @ngreenwald ,

Thanks for your help very much.

Which code was used to generate the validation patches? (from `512x512` to `256x256`)

**Comment by ngreenwald** [comment]
No problem

https://github.com/vanvalenlab/publication-figures/blob/474535fadd1c1c3be9e5294caed6a3ac964ab165/2021-Greenwald_Miller_et_al-Mesmer/dataset_builder.py#L934

If you have any suggestions for updating our documentation to make this more clear, please let me know.

**Comment by JunMa11** [comment]
Hi @ngreenwald ,

The documentation is great. It would be more attractive if there is a resource list of the benchmark.
Basically, one may want to compare new cell segmentation results to the methods that you have evaluated on the official benchmark dataset.

The official [dataset split](https://datasets.deepcell.org/) of the benchmark dataset (npz format) is:
> 512x512 training patches (2601), 256x256 validation patches (3140), 256x256 testing patches (1249)

1. **Could you please share a link to the test set (1249 patches) segmentation results of Cellpose-TN, StarDist-TN, FeatureNet-TN, and RetinaMask? **
> Based on the above dataset split

I have found the segmentation results of the `Mesmer`.
https://storage.googleapis.com/publications-data/mesmer-paper/publication_data_folder.zip
![1642628734(1)](https://user-images.githubusercontent.com/19947331/150218934-436bf57b-dd5e-4c34-8ef9-7e50ce4a2d63.png)

But the segmentation results of **Cellpose** do not correspond to the official dataset split because they are generated with different splits rather than the official datasplits.
![1642629690(1)](https://user-images.githubusercontent.com/19947331/150221175-fcd7a3a4-5077-491b-ae2e-ececc7bf911f.png)


I apologize for so many questions. Thanks for your kindly help.

**Comment by ngreenwald** [comment]
Got it, I think I understand. 

The dataset split that people can download from datasets.deepcell.org is not the exact same as the one we did all the benchmarking on. We had three separate splits for benchmarking, so that we could make replicates. However, even though these splits aren't the "official" dataset split, all of the models we evaluated were evaluated on those splits. Is there a reason using the triplicate splits won't work for your purposes?

**Comment by JunMa11** [comment]
Hi @ngreenwald ,

Thanks for your explanation. I thought the data from datasets.deepcell.org was the recommended split to compare different methods.

Using the triplicate splits works well for my purpose:) Where can I find the corresponding image data and ground truth of the three splits?

**Comment by ngreenwald** [comment]
Hey Jun, 
It depends on what your goal is. As you probably noticed from the plots, we didn't observe much of a difference between the various splits. So having three was useful for the first demonstration that there wasn't much inter-split variation, but for follow-up work, it would probably be okay to use just a single split. 

Given that, it's up to you about which split you would prefer to use. The official TissueNet splits that are downloadable have all the same data as the three splits we used for model evaluation. If you want to wait for us to fix the corrupted zip and use those, that's fine. But it would be equally valid to use the single TissueNet dataset for model comparison/evaluation.

**Comment by JunMa11** [comment]
Hi @ngreenwald ,

Thanks for your reply very much.

My goal is simple: directly compare new segmentation results with the methods that you have evaluated in the paper (without re-running these experiments).

For this goal, I'm looking for
- dataset split
- corresponding segmentation results, e.g., Mesmer, Cellpose, StarDist, FeatureNet

For the official splits, I can download the dataset split but the segmentation results of the Cellpose, StarDist, and FeatureNet are not available because the evaluations in the paper were based on random splits.

For the random splits, I can find the segmentation results now, and I appreciate it very much if you could share the corresponding training data.

**Comment by ngreenwald** [comment]
Got it, I'll regenerate the zip file. Shoot me a message if I haven't replied by Friday

**Comment by JunMa11** [comment]
Hi @ngreenwald ,

I'm sorry to bother you.

Any updates on the zip file?

Kindest regards,
Jun

**Comment by ngreenwald** [comment]
It was uploaded last night, but it looks like the permissions for the link weren't updated. @tddough98 should be able to take care of it monday

**Comment by tddough98** [comment]
Just made the updated data public now! It's available at the same link [https://storage.googleapis.com/publications-data/mesmer-paper/model_training_files.zip](https://storage.googleapis.com/publications-data/mesmer-paper/model_training_files.zip)
